NumberInput component

bcowell opened the pull request:
[Link to ticket](https://app.clubhouse.io/yuhuinc/story/7207/create-number-input-component)

### What are you trying to accomplish?
Create component that is:
- Keyboard accessible
- Tracks value as <input type="number" />
- Min value and max value
- When reached min or max, relevant button is greyed out

### What should reviewers focus on?
Kept the default looks of the input as having a box around it so it's more obvious it's accessible (see images below).

### Photos / Videos
![image](https://user-images.githubusercontent.com/10282189/77656912-ed6f9f80-6f4a-11ea-8b26-4a42f12ef6d3.png)
![image](https://user-images.githubusercontent.com/10282189/77657195-535c2700-6f4b-11ea-8a0d-438bc092394e.png)

fiffty commented on src/Input/NumberInput.tsx:
```diff
@@ -0,0 +1,97 @@
+
+const StyledInner = styled.div`
+  width: 100%;
+  display: inline;
```
Why is this `inline`? I'll add more comments below but let's use flexbox for layouts.
Having this
```css
display: flex;
flex-direction: column;
justify-content: space-between;
```
And its children not have `float` will likely give you the same expected layout.

Good resources for flexbox include:
- https://css-tricks.com/snippets/css/a-guide-to-flexbox/
- https://www.youtube.com/watch?v=K74l26pE4YA

fiffty commented on src/Input/NumberInput.tsx:
```diff
@@ -0,0 +1,97 @@
+  <StyledContainer>
+    <StyledInner>
+      <StyledTitle>
+        <h3 style={{ display: "inline" }}>{title}</h3>
```
Heading elements have semantic meaning and have implications for SEO, accessibility, etc. We should only use them when they are indeed used as "the third level heading".

Let's change this to something like a `div` or a `span`, even a `p` could do.

Also, let's not mix using styled-components and inline-styling as much as possible. Please stick to styled-components if you can. I also don't think `display: inline` is needed here if you go with flexbox.

fiffty commented on src/Input/NumberInput.tsx:
```diff
@@ -0,0 +1,97 @@
+          value={value}
+          onChange={e => {
+            callback(e.target.value);
+          }}
```
Let's add our `min` and `max` value here so that the validation also happens on the browser level.
https://www.w3schools.com/tags/att_input_max.asp

fiffty commented on src/Input/NumberInput.tsx:
```diff
@@ -0,0 +1,97 @@
+  border: ${`solid 1.5px ${colors.PURE_BLUE}`};
+  border-radius: 50%;
+  color: ${colors.PURE_BLUE};
+  opacity: ${props => (props.disabled ? "37%" : "100%")};
```
Do the buttons get `cursor: pointer` by default? That is, when you hover over the button, does the mouse cursor change to the pointy hand?

fiffty commented on src/Input/NumberInput.tsx:
```diff
@@ -0,0 +1,97 @@
+  float: left;
+  font-size: 18px;
+  font-weight: 600;
+  color: ${colors.DARK_GRAY};
```
What happens when `title` is super long? It will likely break the layout. Let's make it so that if it is too long for the width, have it cut off with ellipsis.
https://www.w3schools.com/cssref/css3_pr_text-overflow.asp
https://stackoverflow.com/a/17783233

bcowell replied:
Updated
![image](https://user-images.githubusercontent.com/10282189/77673166-341bc480-6f60-11ea-9283-3ffd34671c09.png)

bcowell commented:
Fixed: 
- Typing in the value will stop at max and min (same with if you use up and down arrows)
- Removed the arrows in the text box as well as the border
- Verified you can press navigate with tab and shift+tab to go from buttons to the input element
